### PR TITLE
feat: init wizard v2 improvements

### PR DESCRIPTION
**Pull request by aorumbayev**
## Proposed Changes

- Refining `presets`, adding notion of a custom `preset`
- Reusing algokit-base-template for workspace specific artifacts - to remove coupling and redundancy
- Renaming `templates_generated` to more generic `examples` folder that by default tests the main 'starter' and 'production' presets
- Adding `Contribution` guide and simple starting point for generating dummy instances of templates for quick preview during development
- Removing dependency on beaker, instead allows specifying all main frameworks to pair against default react component
- Refining codespaces setup for vscode for puya, beaker. Tealscript is currently not supporting the full e2e launch config flow hence template will only supply base codespaces file for aggregate folder preview (no custom launch configs). 
- Fixing visitors badge
- Base template at the workspace level (see step 1 in _tasks in copier.yaml at the root) - also drops a devcontainer artifact for github codespaces (similar behaviour for all standalone templates too). Making it easy to quickly load up any workspace typed algokit project to a dev environment from their github repo. A feature heavily utilized by DevRels during various bootcamp activities, given existing feedback and feature requests as part of the wizard adr **and** small footprint i think its a fair point to keep devcontainers at the base template level.
- TEMPORARY - changing pipx install algokit to version from wizard v2 branch
- ~~TEMPORARY - ci is to be fixed as soon as base template repo becomes public~~ -> done

### CODE REVIEW

**Comment by aorumbayev** [comment]
@neilcampbell for simpler review, ignore changes in tests_generated the main gist is in template_content and copier.yaml, and examples could probably checked last. 

I could rename examples back to tests_generated folder but given the presets being changed to hide all questions by default, there would have been a lot of artifacts in diff regardless since number of default scenarios is now reduced to just 2 options. The workspace structure would cause changes within each artifact as well so renaming to `examples` is marginally similar in size. As for why its set to examples, see a note in README.md that mentions that this folder can be used for quick preview of sample pre-made projects.

File: template_content/{% if use_github_actions %}.github{% endif %}/workflows/frontend-ci.yaml [hunk 1]
```diff
@@ -1,63 +1,63 @@
 name: Frontend CI
```
Comment (neilcampbell) [inline template_content/{% if use_github_actions %}.github{% endif %}/workflows/frontend-ci.yaml:1]: Needs the `.jinja` extension.

File: template_content/inject_content/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## How to add new Algorand smart contracts into the frontend project?
+
+The following folder is reserved for the Algorand Application Clients. The clients are used to interact with instances of Algorand Smart Contracts (ASC1s) deployed on-chain.
+
+When you initially create your project the frontend starter automatically compiles (see [`generate:app-clients`](../../package.json)) the default `{{ contract_name }}` contract as `{{ contract_name.split('_')|map('capitalize')|join }}Client` and provides the [`AppCalls.tsx`](../components/AppCalls.tsx) component showcasing how to interact with that application client.
+
+When you create new Beaker contracts on the backend folder and build them (so that they are available under `artifacts` folder) - simply run `npm run build` to automatically:
+
+1. generate new typed clients based on all contracts currently available under `artifacts` folder.
+2. move them to `frontend/src/contracts` folder.
```
Comment (neilcampbell) [inline template_content/inject_content/README.md:10]:
```suggestion
2. Copy the typed clients into this folder (`src/contracts`).
```

File: template_content/inject_content/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## How to add new Algorand smart contracts into the frontend project?
+
+The following folder is reserved for the Algorand Application Clients. The clients are used to interact with instances of Algorand Smart Contracts (ASC1s) deployed on-chain.
+
+When you initially create your project the frontend starter automatically compiles (see [`generate:app-clients`](../../package.json)) the default `{{ contract_name }}` contract as `{{ contract_name.split('_')|map('capitalize')|join }}Client` and provides the [`AppCalls.tsx`](../components/AppCalls.tsx) component showcasing how to interact with that application client.
+
+When you create new Beaker contracts on the backend folder and build them (so that they are available under `artifacts` folder) - simply run `npm run build` to automatically:
+
+1. generate new typed clients based on all contracts currently available under `artifacts` folder.
```
Comment (neilcampbell) [inline template_content/inject_content/README.md:9]:
```suggestion
1. Generate new typed clients based on all contracts currently available under `artifacts` folder.
```

File: template_content/inject_content/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+## How to add new Algorand smart contracts into the frontend project?
+
+The following folder is reserved for the Algorand Application Clients. The clients are used to interact with instances of Algorand Smart Contracts (ASC1s) deployed on-chain.
+
+When you initially create your project the frontend starter automatically compiles (see [`generate:app-clients`](../../package.json)) the default `{{ contract_name }}` contract as `{{ contract_name.split('_')|map('capitalize')|join }}Client` and provides the [`AppCalls.tsx`](../components/AppCalls.tsx) component showcasing how to interact with that application client.
```
Comment (neilcampbell) [inline template_content/inject_content/README.md:5]:
```suggestion
When you initially create your project, the frontend starter automatically compiles (see [`generate:app-clients`](../../package.json)) the default `{{ contract_name }}` contract as `{{ contract_name.split('_')|map('capitalize')|join }}Client` and provides the [`AppCalls.tsx`](../components/AppCalls.tsx) component showcasing how to interact with that application client.
```

**Comment by aorumbayev** [comment]
@neilcampbell feel free to unresolve any of the above but should be now addressed. Will check out the comments on react

**Review by neilcampbell — APPROVED** [review]

**Pull request merged**